kroxylicious-api/src/main/java/io/kroxylicious/proxy/service/Contributor.java

```diff
@@ -23,6 +25,20 @@ public interface Contributor<T> {
      */
     Class<? extends BaseConfig> getConfigType(String shortName);
 
+    /**
+     * Used to determine if a configuration object is required for this filter.
+     * By default, it is assumed that if a custom config type is specified it is required.
+     * To make custom config classes entirely optional implementors should override this method.
+     *
+     * @param shortName service short name
+     *
+     * @return <code>true</code> if the configuration must be specified.
+     */
+    default boolean requiresConfig(String shortName) {
```

Comment:
Think it needs to be Boolean and return `null` if there is a null config type (because this Contributor doesn't offer that shortName)

Reply:
could be a possible benefit of refactoring away from using multiple methods, if we could get to something like:

```
record Factory<T>(Class<? extends BaseConfig> configClass, boolean requiresConfig) {
    T getInstance(BaseConfig config);
}

Contributor<T> {
    getFactory(String shortName); // returns null if Contributor doesn't support this shortname
}
```
Then we'd have one place where nullness means "I don't support that shortName".